### PR TITLE
Pickup items with harm enabled

### PR DESCRIPTION
**Pull request by RavMorgan**
## About the PR

Added ability to pick up items in harm mode

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://user-images.githubusercontent.com/48182970/229576855-bd3fb097-e034-4919-b60f-0a8d59851fc7.mp4

**Changelog**

:cl:
- add: Added ability to pick up items in harm mode!

### CODE REVIEW

**Comment by Chudjak** [comment]
linter failing because holy shit I hate git, it is not related to your PR.

**Comment by RavMorgan** [comment]
I know.

File: Content.Shared/Interaction/SharedInteractionSystem.cs [hunk 1]
```diff
@@ -253,6 +254,19 @@ private bool ShouldCheckAccess(EntityUid user)
...
+                if(!TryComp<ItemComponent>(target.Value, out _)) return;
+
+                // We need this because of funny SharedGunSystem realization
+                if (TryComp<GunComponent>(target.Value, out var gunComponent))
+                {
+                    gunComponent.NextFire = _gameTiming.CurTime + TimeSpan.FromMilliseconds(125);
+                    Dirty(gunComponent);
+                }
+
+                InteractHand(user, target.Value);
```
Comment (Chudjak) [inline Content.Shared/Interaction/SharedInteractionSystem.cs:268]: Before I merge it, why exactly was this needed?

Comment (RavMorgan) [reply Content.Shared/Interaction/SharedInteractionSystem.cs:268]: If this is not added, the weapon will fire as soon as you pick it up

Comment (Chudjak) [reply Content.Shared/Interaction/SharedInteractionSystem.cs:268]: Ah, I see. Thanks.

**Pull request merged**